src/cobra/core/gene.py

```diff
@@ -181,10 +183,9 @@ def parse_gpr(str_expr: str) -> Tuple:
     Use GPR(string_gpr=str_expr) in the future. Because of the GPR() class,
     this function will be removed.
     """
-    warn(
+    logger.warning(
```

Comment:
I think an actual deprecation warning was actually the right thing to do here. What made you change your mind?

Reply:
I was under the impression that logger.warning should be used instead of warning. I'll change it back.

Reply:
I don't have a good rule for when to use which but to my mind the `DeprecationWarning` and `SyntaxWarning` are good cases to raise as objects.

Reply:
The SyntaxWarning I have it as both logging and warning, to be thorough.